src/Pim/Bundle/CatalogBundle/Doctrine/ORM/CompletenessRemover.php

```diff
@@ -87,33 +88,44 @@ public function removeForChannelAndLocale(ChannelInterface $channel, LocaleInter
     {
         $products = $this->createProductQueryBuilder($channel, $locale)->execute();
 
-        $this->bulkRemoveCompletenesses($products);
+        $this->bulkRemoveCompletenesses($products, $channel, $locale);
     }
 
     /**
-     * Drops the current completenesses from the database and from the index for a list of products.
+     * Drops the current completenesses from the database and from the index for
+     * a list of products and optionally for a channel and/or locale.
      *
-     * @param CursorInterface $products
+     * @param CursorInterface       $products
+     * @param null|ChannelInterface $channel
+     * @param null|LocaleInterface  $locale
      */
-    protected function bulkRemoveCompletenesses(CursorInterface $products)
-    {
-        $statement = $this->entityManager->getConnection()->prepare('
-            DELETE c
-            FROM pim_catalog_completeness c
-            WHERE c.product_id IN (:productIds)
-        ');
-
+    protected function bulkRemoveCompletenesses(
+        CursorInterface $products,
+        ChannelInterface $channel = null,
+        LocaleInterface $locale = null
+    ) {
         $bulkedProducts = [];
         $productIds = [];
         $bulkCounter = 0;
 
+        $query = 'DELETE c FROM pim_catalog_completeness c WHERE c.product_id IN (?)';
+        if (null !== $channel) {
+            $query .= sprintf(' AND c.channel_id = %d', $channel->getId());
```

Comment:
please don't do that :) never again :)
even if here, it's an integrer, you're are not sure about what you get
you are not sure about what your teammates will do in 6 months, when they'll get back here to fix a bug

use the execute query method to escape the values